src/ir/objc.rs

```diff
@@ -0,0 +1,93 @@
+use clang;
+use clang_sys::CXChildVisit_Continue;
+use clang_sys::CXCursor_ObjCInstanceMethodDecl;
+use clang_sys::CXCursor_ObjCInterfaceDecl;
+use clang_sys::CXCursor_ObjCProtocolDecl;
+use parse::{ClangSubItemParser, ParseError, ParseResult};
+// use parse_one;
+// use clang_sys::CXCursor_ObjCSuperClassRef;
+use super::context::BindgenContext;
+
+#[derive(Debug)]
+pub struct ObjCInterface {
+    pub name: String,
+    pub methods: Vec<ObjCInstanceMethod>,
+}
+
+#[derive(Debug)]
+pub struct ObjCInstanceMethod {
+    pub name: String, // signature
+    pub rust_name: String,
+}
+
+impl ObjCInterface {
+    fn new(name: &str) -> ObjCInterface {
+        ObjCInterface {
+            name: name.to_owned(),
+            methods: Vec::new(),
+        }
+    }
+}
+
+impl ClangSubItemParser for ObjCInterface {
+    fn parse(cursor: clang::Cursor,
+             _ctx: &mut BindgenContext)
+             -> Result<ParseResult<Self>, ParseError> {
+
+
+        match cursor.kind() {
+            CXCursor_ObjCInterfaceDecl |
+            CXCursor_ObjCProtocolDecl => {
```

Comment:
Hmmm, does libclang return a specific `CXTypeKind` for these (you can see this printing `cursor.cur_type()`)? If so, then merging it into `Type` is probably the best path forward, otherwise, it'll probably need a bit of thinking to get it right.